nf_core/download.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python
+""" Download a nf-core pipeline """
+
+from __future__ import print_function
+
+from io import BytesIO
+import logging
+import os
+import requests
+import subprocess
+import sys
+from zipfile import ZipFile
+
+
+import nf_core.list, nf_core.utils
+
+class DownloadWorkflow():
+
+    def __init__(self, pipeline, release=None, singularity=False, outdir=None):
+        """ Set class variables """
+
+        self.pipeline = pipeline
+        self.release = release
+        self.singularity = singularity
+        self.outdir = outdir
+
+        self.wf_name = None
+        self.wf_sha = None
+        self.wf_download_url = None
+        self.config = dict()
+        self.containers = list()
+
+    def download_workflow(self):
+        """ Main function to download a nf-core workflow """
+
+        # Get workflow details
+        if not self.get_workflow():
+            sys.exit(1)
+
+        # Check that the outdir doesn't already exist
+        if os.path.exists(self.outdir):
+            logging.error("Output directory '{}' already exists".format(self.outdir))
+            sys.exit(1)
+
+        logging.info(
+            "Saving {}".format(self.pipeline) +
+            "\n Pipeline release: {}".format(self.release) +
+            "\n Pull singularity containers: {}".format('Yes' if self.singularity else 'No') +
+            "\n Output directory: {}".format(self.outdir)
+        )
+
+        # Download the pipeline files
+        logging.info("Downloading workflow files from GitHub")
+        self.download_wf_files()
+
+        # Download the singularity images
+        if self.singularity:
+            logging.info("Fetching container names for workflow")
+            self.find_singularity_images()
+            if len(self.containers) == 0:
+                logging.info("No container names found in workflow")
+            else:
+                os.mkdir(os.path.join(self.outdir, 'singularity-images'))
+                for container in self.containers:
+                    self.download_singularity_image(container)
+
+
+    def get_workflow(self):
```

Comment:
Trying to be picky 😉 -> is it really a *"get"* function ? I mean it returns only `True/False`. So maybe naming `set_workflow` or `collect_workflow` or something else be more appropriate

Reply:
Alright, I'll give you this one ;)